pkg/test/step.go

```diff
@@ -412,6 +416,28 @@ func (s *Step) CheckAssertCommands(ctx context.Context, namespace string, comman
 	return testErrors
 }
 
+func (s *Step) CheckAssertExpressions(namespace string) []error {
+	client, err := s.Client(false)
+	if err != nil {
+		return []error{err}
+	}
+
+	variables := make(map[string]interface{})
+	for _, resourceRef := range s.Assert.ResourceRefs {
+		if resourceRef.Namespace == "" {
+			resourceRef.Namespace = namespace
+		}
+		namespacedName, referencedResource := resourceRef.BuildResourceReference()
+		if err := client.Get(context.TODO(), namespacedName, referencedResource); err != nil {
+			return []error{fmt.Errorf("failed to get referenced resource '%v': %w", namespacedName, err)}
+		}
+
+		variables[resourceRef.Ref] = referencedResource.Object
+	}
```

Comment:
This should be lazy loaded.

Reply:
@eddycharly you mean as a performance improvement? Or do you see a correctness issue?

Reply:
Somehow both. If the first condition is met in `any` why bother evaluating other conditions ?
I can imagine cases where the first condition returns true but the second condition throws an error. Returning early makes the evaluation succeed.

Reply:
(this may or may not be what you want though)